infinite-scroll/guide.md

```diff
@@ -0,0 +1,414 @@
+    index = helper.client.initIndex(this.args.state.index);
+    hitsDiv.removeEventListener('scroll', searchNewRecords.bind(this));
+    hitsDiv.addEventListener('scroll', browseNewRecords.bind(this));
+    addBrowsedRecords.call(this);
+  }
+};
+```
+
+That's a lot to take in all at once, so let's go through it one-by-one.
+
+```
+var cursor, index, page, nbPages, loading;
+
+var hitsDiv = document.getElementById('hits');
+```
+
+We're creating variables outside of our widget or our functions because we need to access and update these in a number of different places (e.g. `init`, `render`) and dependency injection isn't always feasible. Be careful to not make these global to your app.
+
+```
+var scrolledNearBottom = function(el){
+  return (el.scrollHeight - el.scrollTop) < 850;
+};
+
+var searchNewRecords = function(){
+  if(scrolledNearBottom(hitsDiv)) {
+    addSearchedRecords.call(this);
+  }
+};
+```
+
+As the user scrolls, we will be checking to see if they are near enough to the bottom that we need to grab new results. We've set this to return `true` when the user is 850 pixels from the bottom, but you can change this to your needs.
+
+```
+var addSearchedRecords = function(){
+  if(!loading && page < nbPages - 1) {
+    loading = true;
+    page += 1;
+    helper.searchOnce({page: page}, appendSearchResults.bind(this));
+  }
+};
+
+var appendSearchResults = function(err, res, state){
+  page = res.page;
+  _.assign(res, {pageNo: page + 1});
+  loading = false;
+
+  var result = renderTemplate(this.templates.items, res);
+  this.container.appendChild(result);
+
+  if(page === nbPages - 1 && (this.args.results.nbHits > nbPages * this.args.results.hitsPerPage)){
+    index = helper.client.initIndex(this.args.state.index);
+    hitsDiv.removeEventListener('scroll', searchNewRecords.bind(this));
+    hitsDiv.addEventListener('scroll', browseNewRecords.bind(this));
+    addBrowsedRecords.call(this);
+  }
+};
+```
+
+Next up, we want to check two things: that the next set of results isn't loading and that we aren't at the end.
+
+If both of those are the case, we signal that the results are loading and search for results, specifying the page and using the algolia search helper. When the results have returned, the helper will call our `appendSearchResults` callback.
+
+That callback is pretty simple: we take our results, add a `pageNo` property to the object, and pass it to mustache. The last part deserves a closer look. If:
+
+- We have reached the final page **and**
+- The total number of hits for that search is greater than the number we've displayed
+
+**Then** we need to switch from searching our records to browsing them.
+
+First, a detour.
+
+## Searching vs Browsing
+
+One thing we need to know is that Algolia will return at most the first 1,000 results for any search. This allows us to return results quickly and with maximum relevancy. In nearly all settings, you won't need more than 1,000 results.
```

Comment:
I personally wouldn't take the first form as condescending at all (I'm reading to learn) but your other proposal works for me :+1: